Projects/Backendless.NET.Standard20/NewHttpEngine.cs

```diff
@@ -0,0 +1,226 @@
+                                  IDictionary messageHeaders, IDictionary httpHeaders, Responder<T> responder,
+                                  AsyncStreamSetInfo<T> asyncStreamSetInfo ) =>
+      SendRequest( CreateMessageForInvocation( className, methodName, args, messageHeaders ),
+                                requestHeaders, httpHeaders, responder, asyncStreamSetInfo );
+
+#if !( NET_35 || NET_40 )
+    public override Task<T> SendRequest<T>( V3Message v3Msg, IDictionary requestHeaders, IDictionary httpHeaders,
+                                                                  ResponseThreadConfigurator threadConfigurator ) =>
+      SendHttpRequest<T>( v3Msg, requestHeaders, httpHeaders, threadConfigurator );
+
+    internal override Task<T> Invoke<T>( string className, string methodName, object[] args, IDictionary requestHeaders,
+               IDictionary messageHeaders, IDictionary httpHeaders, ResponseThreadConfigurator threadConfigurator ) =>
+      SendRequest<T>( CreateMessageForInvocation( className, methodName, args, messageHeaders ),
+                                        requestHeaders, httpHeaders, threadConfigurator );
+#endif
+
+    private async void SendHttpRequest<T>( V3Message v3Msg, IDictionary requestHeaders, IDictionary httpHeaders, Responder<T> responder, AsyncStreamSetInfo<T> asyncStreamSetInfo )
+    {
+      byte[] requestBytes = CreateRequest( v3Msg, requestHeaders );
+      HttpRequestMessage requestMessage = new HttpRequestMessage
+      {
+        RequestUri = new Uri( GatewayUrl ),
+        Method = HttpMethod.Post,
+        Content = new ByteArrayContent( requestBytes )
+      };
+
+      requestMessage.Content.Headers.ContentType = new MediaTypeHeaderValue( "application/x-amf" );
+      HttpResponseMessage responseMessage = await httpClient.SendAsync( requestMessage );
+
+      asyncStreamSetInfo.responder = responder;
+
+      var streamResponse = await responseMessage.Content.ReadAsStreamAsync();
+      var curTime = DateTime.Now.Ticks;
+      var roundTrip = ( curTime - asyncStreamSetInfo.messageSentTime ) / TimeSpan.TicksPerMillisecond;
+      var parser = new RequestParser();
+      var responseObject = parser.readMessage( streamResponse );
+      var responseData = (object[]) responseObject.getRequestBodyData();
+      var v3 = (V3Message) ( (IAdaptingType) responseData[ 0 ] ).defaultAdapt();
+
+      if( v3.isError )
+      {
+        var errorMessage = (ErrMessage) v3;
+        var fault = new Fault( errorMessage.faultString, errorMessage.faultDetail, errorMessage.faultCode );
+        asyncStreamSetInfo.responder?.ErrorHandler( fault );
+        return;
+      }
+
+      var body = (IAdaptingType) ( (AnonymousObject) ( (NamedObject) responseData[ 0 ] ).TypedObject ).Properties[ "body" ];
+      var result = (T) body.adapt( typeof( T ) );
+
+      asyncStreamSetInfo.responder?.ResponseHandler( result );
+    }
+
+#if !( NET_35 || NET_40 )
```

Comment:
how about splitting old approach and new approach into 2 separate classes?

Reply:
That was done